clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java

```diff
@@ -64,66 +73,117 @@ public PartitionData(long timestamp, int maxNumOffsets) {
         }
     }
 
+    /**
+     * Constructor for ListOffsetRequest v0
+     */
+    @Deprecated
     public ListOffsetRequest(Map<TopicPartition, PartitionData> offsetData) {
-        this(CONSUMER_REPLICA_ID, offsetData);
+        this(CONSUMER_REPLICA_ID, offsetData, 0);
     }
 
+    /**
+     * Constructor for ListOffsetRequest v0
+     */
+    @Deprecated
     public ListOffsetRequest(int replicaId, Map<TopicPartition, PartitionData> offsetData) {
-        super(new Struct(CURRENT_SCHEMA));
-        Map<String, Map<Integer, PartitionData>> topicsData = CollectionUtils.groupDataByTopic(offsetData);
+        this(replicaId, offsetData, 0);
+    }
+
+    /**
+     * Constructor with a specified version.
+     */
+    public ListOffsetRequest(Map<TopicPartition, ?> targetTimes, int version) {
```

Comment:
The request classes are actually not public classes. The only public classes are the ones listed in the javadoc: https://kafka.apache.org/0100/javadoc/index.html?org/apache/kafka/clients/consumer/KafkaConsumer.html